### PR TITLE
Latest headers Pub-Sub

### PR DESCRIPTION
**Pull request by tomusdrw**
Related #884 

The first subscription from geth-compatible interface.

### CODE REVIEW

File: rpc/src/v1/impls/eth_pubsub.rs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+		_proposed: Vec<Bytes>,
+		_duration: u64,
+	) {
+		const EXTRA_INFO_PROOF: &'static str = "Object exists in in blockchain (fetched earlier), extra_info is always available if object exists; qed";
+		let blocks = enacted
+			.into_iter()
+			.filter_map(|hash| self.client.block_header(BlockId::Hash(hash)))
+			.map(|header| {
+				let hash = header.hash();
+				(header, self.client.block_extra_info(BlockId::Hash(hash)).expect(EXTRA_INFO_PROOF))
```
Comment (rphmeier) [inline rpc/src/v1/impls/eth_pubsub.rs]: headers are pruned out of the light client very quickly, so I wouldn't expect this to always pass.

Comment (tomusdrw) [inline rpc/src/v1/impls/eth_pubsub.rs]: That's an implementation for full client, light implementation doesn't have `extra_info` at all.

**Comment by rphmeier** [comment]
What about handling `retracted` blocks for reorg? Does the geth RPC not do that?

**Comment by tomusdrw** [comment]
In case of re-org `enacted` should contain all the blocks that become canonical, the client will receive headers with numbers that were received in the past. It should be the same behaviour as for geth.

**Pull request merged**